examples/llm_compression/torch/qat_with_nls_downstream/main.py

```diff
@@ -574,22 +640,44 @@ def main(argv) -> float:
     train_dataset = [tokenize(tokenizer, sample) for sample in train_dataset]
     random.shuffle(train_dataset)
 
-    model = compress_weights(
-        model,
-        dataset=Dataset([{k: v.to(device) for k, v in model_input.items()}]),
-        **compression_config,
-    )
-    results_of_compressed_model = lm_eval(model, task=args.task, batch_size=args.eval_batch_size, tokenizer=tokenizer)
-    print(f"Results of NNCF compressed model={json.dumps(results_of_compressed_model, indent=4)}")
-    overall_result["results_of_compressed_model"] = results_of_compressed_model
-    initial_result = results_of_compressed_model[args.lm_eval_metric]
-
     # Create or load model to tune with Fake Quantizers and absorbable LoRA adapters.
     if args.resume and ckpt_file.exists():
-        model = AutoModelForCausalLM.from_pretrained(args.pretrained, torch_dtype=torch_dtype, device_map=device)
+        # Initial results (before fine-tuning) are needed to compare the improvements after fine-tuning.
+        # If the previous results are not saved, we will get them again and should use the compressed model
+        # without fine-tuning instead of the resume model.
+        if "torch_result_compression_before_finetuning" not in overall_result:
+            with tempfile.TemporaryDirectory() as tmpdir:
+                tmp_ckpt_file = Path(tmpdir) / "temp_ckpt.pth"
+                tmp_model = copy.deepcopy(model)
+                tmp_model = compress_weights(
+                    tmp_model,
+                    dataset=Dataset([{k: v.to(device) for k, v in tmp_model.dummy_inputs.items()}]),
+                    **compression_config,
+                )
+                save_checkpoint(tmp_model, tmp_ckpt_file)
+                with create_eval_model(
+                    tmp_model, args.fast_eval, args.pretrained, torch_dtype, tmp_ckpt_file
+                ) as eval_model:
+                    results_compression_before_finetuning = lm_eval(
+                        eval_model, task=args.task, batch_size=args.eval_batch_size
+                    )
+                overall_result["torch_result_compression_before_finetuning"] = results_compression_before_finetuning
```

Comment:
Is this part really needed for customers? In my opinion, it adds extra complexity 
It also creates 2 copies of the model `tmp_model = copy.deepcopy(model)` and inside `create_eval_model`.

Reply:
Thanks for the suggestions, @ljaljushkin ! Please check the latest commit addressing this issue.